src/core/api/user.js

```diff
@@ -0,0 +1,12 @@
+/* @flow */
+import { callApi } from 'core/api';
+import type { ApiStateType } from 'core/reducers/api';
+
+
+export function userProfile({ api }: {| api: ApiStateType |}) {
```

Comment:
It would be pretty confusing if the caller forgot to pass a value for `api` so I think this should throw an error if it's empty.

Reply:
can we open an issue for that? this should be done across all `core/api/*` functions.

Reply:
can we open an issue for that? this should be done across all `core/api/*` functions.

Reply:
I don't think it should be done for all API functions. In this case, it's only necessary to require `api` because you have to be authenticated to make a request to the profile endpoint.

Reply:
ok, adding the check here then